commit 8db49cc2de52033990ae5d4d6aacedc8f986e803
Author: Will Newton <will.newton@linaro.org>
Date:   Thu Oct 31 17:21:11 2013 -0700

    config/tc-aarch64.c: Avoid trying to parse a vector mov as immediate.
    
    Parsing a vector mov instruction currently leads to a phantom undefined
    symbol being added to the symbol table. e.g.:
    
           .text
           mov     x0, v0.D[0]
    
    Produces an undefined symbol called "v0.D".
    
    gas/ChangeLog:
    
    2013-11-05  Will Newton  <will.newton@linaro.org>
    
    	PR gas/16103
    	* config/tc-aarch64.c (parse_operands): Avoid trying to
    	parse a vector register as an immediate.
    
    gas/testsuite/ChangeLog:
    
    2013-11-05  Will Newton  <will.newton@linaro.org>
    
    	* gas/aarch64/advsimd-mov-bad.d: New file.
    	* gas/aarch64/advsimd-mov-bad.s: Likewise.


diff --git a/gas/config/tc-aarch64.c b/gas/config/tc-aarch64.c
index 14ffdad..02fe4de 100644
--- a/gas/config/tc-aarch64.c
+++ b/gas/config/tc-aarch64.c
@@ -4810,7 +4810,8 @@ parse_operands (char *str, const aarch64_opcode *opcode)
 	case AARCH64_OPND_IMM_MOV:
 	  {
 	    char *saved = str;
-	    if (reg_name_p (str, REG_TYPE_R_Z_SP))
+	    if (reg_name_p (str, REG_TYPE_R_Z_SP) ||
+		reg_name_p (str, REG_TYPE_VN))
 	      goto failure;
 	    str = saved;
 	    po_misc_or_fail (my_get_expression (&inst.reloc.exp, &str,
diff --git a/gas/testsuite/gas/aarch64/advsimd-mov-bad.d b/gas/testsuite/gas/aarch64/advsimd-mov-bad.d
new file mode 100644
index 0000000..6ca9887
--- /dev/null
+++ b/gas/testsuite/gas/aarch64/advsimd-mov-bad.d
@@ -0,0 +1,12 @@
+#source: advsimd-mov-bad.s
+#readelf: -s --wide
+
+Symbol table '.symtab' contains 6 entries:
+ +Num:.*
+ +[0-9]+:.*
+ +[0-9]+:.*
+ +[0-9]+:.*
+ +[0-9]+:.*
+ +[0-9]+:.*
+#failif
+ +[0-9]+: +[0-9a-f]+ +0 +NOTYPE +GLOBAL +DEFAULT +UND v0.D
diff --git a/gas/testsuite/gas/aarch64/advsimd-mov-bad.s b/gas/testsuite/gas/aarch64/advsimd-mov-bad.s
new file mode 100644
index 0000000..d7ba226
--- /dev/null
+++ b/gas/testsuite/gas/aarch64/advsimd-mov-bad.s
@@ -0,0 +1,2 @@
+        .text
+        mov     x0, v0.D[0]
